connectors/pdepend.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import os
+import subprocess
+import tempfile
+from os.path import exists
+import xml.etree.ElementTree as ET
+
+from models.metric import Metric
+from utils.math import Math
+from utils.timeit import timeit
+
+
+class PDependConnector:
+    """
+    Connector to PHP Depend CLI tool
+    https://github.com/pdepend/pdepend
+
+    Attributes:
+    -----------
+        - directory   Full path to a cloned GIT repository
+        - session     Connection to a database managed by sqlalchemy
+        - version     Sqlalchemy object representing a Version
+    """
+    def __init__(self, directory, version, session, config):
+        self.directory = directory
+        self.session = session
+        self.version = version
+        self.configuration = config
+
+    def analyze_source_code(self):
+        """
+        Analyze the repository by using PHP Depend analysis tool
+        """
+        logging.info('PDepend::analyze_repo')
+        # Test if metrics have been already generated for this version
+        metric = self.session.query(Metric).filter(Metric.version_id == self.version.version_id).first()
+        if not metric:
+            metric = Metric()
+        if self.configuration.language != "PHP":
+            logging.info('PDepend is only used for PHP language')
+        # If the WMC PDepend metric is not present, we assume the analysis hasn't been done for this version, so we do it
+        elif not metric.pdepend_wmc:
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                report_path = self.generate_report(tmp_dir)
+                metric = self.compute_metrics(metric, report_path)
+            self.save_metrics_to_db(metric)
+        else:
+            logging.info('PDepend analysis already done for this version')
+
+    def _xml_get_nodes_atribute(self, root, node_name, attribute_name):
+        nodes = root.findall(f".//{node_name}[@{attribute_name}]")
+        return [node.attrib[attribute_name] for node in nodes]
+    
+    def _list_str_to_int(self, list_):
+        return list(map(int, list_))
+    
+    def __compute_mean(self, xml_root, metric, mean_divide):
+        return Math.get_rounded_mean(
+            self._list_str_to_int(
+                self._xml_get_nodes_atribute(xml_root, mean_divide, metric)
```

Comment:
What if the list is empty -> there is a function in util/math.py now -> get_mean_safe

Reply:
Done